components/org.wso2.carbon.identity.api.server.idp/org.wso2.carbon.identity.api.server.idp.v1/src/main/java/org/wso2/carbon/identity/api/server/idp/v1/core/ServerIdpManagementService.java

```diff
@@ -1758,42 +1764,52 @@ private MetaOutboundConnector createMetaOutboundConnector(ProvisioningConnectorC
     }
 
     private void updateFederatedAuthenticatorConfig(IdentityProvider idp, FederatedAuthenticatorRequest
-            federatedAuthenticatorRequest, boolean isNewFederatedAuthenticator) {
+            federatedAuthenticatorRequest, boolean isNewFederatedAuthenticator)
+            throws IdentityProviderManagementClientException {
 
         if (federatedAuthenticatorRequest != null) {
             List<FederatedAuthenticator> federatedAuthenticators = federatedAuthenticatorRequest.getAuthenticators();
             String defaultAuthenticator = federatedAuthenticatorRequest.getDefaultAuthenticatorId();
             FederatedAuthenticatorConfig defaultAuthConfig = null;
             List<FederatedAuthenticatorConfig> fedAuthConfigs = new ArrayList<>();
             for (FederatedAuthenticator authenticator : federatedAuthenticators) {
-                FederatedAuthenticatorConfig authConfig = new FederatedAuthenticatorConfig();
-                authConfig.setName(base64URLDecode(authenticator.getAuthenticatorId()));
-                authConfig.setDisplayName(getDisplayNameOfAuthenticator(authConfig.getName()));
-                authConfig.setEnabled(authenticator.getIsEnabled());
-
+                String authenticatorName = getDecodedAuthName(authenticator.getAuthenticatorId());
```

Comment:
Shall we rename as getDecodedAuthenticatorName

Reply:
Addressed.